### PR TITLE
Amr/wavemeter rear port

### PR DESCRIPTION
**Pull request by aransfor**
Sorry about the default user. I will update the computer. @gregllong @theoriginaljuice @jayich I have done two things with this pull request, I added a minor function to the wavemeter server and I have mimicked the martinis serial server for the evPump. The advantages of this format is multiple devices can be immediately used. Also serial server, device server and the client will run independent of  each other so startup order does not matter.

### CODE REVIEW

File: lib/clients/evPump/evPumpClient.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def update_power(self, c, power):
         self.powerprogbar.setFormat(str(power['W']) + 'W')
 
     def update_temperature(self, c, temperature):
+        print temperature
```
Comment (jayich) [inline lib/clients/evPump/evPumpClient.py]: debug print statement?

File: lib/servers/evPump/evPumpServer.py [hunk 2]
```diff
@@ -15,33 +15,79 @@
...
+class evPumpDevice(DeviceWrapper):
+
+    @inlineCallbacks
+    def connect(self, server, port):
+        """Connect to a evPump device."""
+        print 'connecting to "%s" on port "%s"...' % (server.name, port),
+        self.server = server
+        self.ctx = server.context()
+        self.port = port
+        p = self.packet()
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: might as well just call "p" "packet", looks like there aren't space issues.

File: lib/servers/evPump/evPumpServer.py [hunk 3]
```diff
@@ -15,33 +15,79 @@
...
+        """Create a packet in our private context."""
+        return self.server.packet(context=self.ctx)
+
+    def shutdown(self):
+        """Disconnect from the serial port when we shut down."""
+        return self.packet().close().send()
+
+    @inlineCallbacks
+    def write(self, code):
+        """Write a data value to the heat switch."""
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: docstring update

File: lib/servers/evPump/evPumpServer.py [hunk 4]
```diff
@@ -15,33 +15,79 @@
...
+        yield self.packet().write_line(code).send()
+
+    @inlineCallbacks
+    def query(self, code):
+        """ Write, then read. """
+        p = self.packet()
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: p -> packet, or even serial_packet

File: lib/servers/evPump/evPumpServer.py [hunk 5]
```diff
@@ -15,33 +15,79 @@
...
+        p.write_line(code)
+        p.read_line()
+        ans = yield p.send()
+        returnValue(ans.read_line)
+
+    @inlineCallbacks
+    def main_loop(self):
+        temp = yield self.query('?T')
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: temp -> temperature

File: lib/servers/evPump/evPumpServer.py [hunk 6]
```diff
@@ -53,110 +99,126 @@ class eVPump(SerialDeviceServer):
...
+    def loadConfigInfo(self):
+        """Load configuration information from the registry."""
+        reg = self.reg
+        yield reg.cd(['', 'Servers', 'evPump', 'Links'], True)
+        dirs, keys = yield reg.dir()
+        p = reg.packet()
+        for k in keys:
+            p.get(k, key=k)
+        ans = yield p.send()
+        self.serialLinks = dict((k, ans[k]) for k in keys)
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: maybe serialLinks should be serial_dict?  I'm not really sure what it does.

File: lib/servers/evPump/evPumpServer.py [hunk 7]
```diff
@@ -53,110 +99,126 @@ class eVPump(SerialDeviceServer):
...
+        """Find available devices from list stored in the registry."""
+        devs = []
+        for name, (serServer, port) in self.serialLinks.items():
+            if serServer not in self.client.servers:
+                continue
+            server = self.client[serServer]
+            ports = yield server.list_serial_ports()
+            if port not in ports:
+                continue
+            devName = '%s - %s' % (serServer, port)
```
Comment (jayich) [inline lib/servers/evPump/evPumpServer.py]: device_name

**Comment by aransfor** [comment]
@gregllong the path to the connected ports is registry/Servers/evPump/Links

the key value pair looks like this: ("qsimexpcontrol Serial Server", "/dev/ttyUSB0")

**Comment by gregllong** [comment]
The evpump now works for the molecule experiment. LGTM

**Comment by gregllong** [comment]
BTW how do you approve a PR? I want that!

**Comment by theoriginaljuice** [comment]
LGTM

**Comment by aransfor** [comment]
@jayich @gregllong @theoriginaljuice 

This pull request has been a pain and deviated from its original intent... Just look over these files that fix serial connection issues and I will merge and use git in a more responsible way in the future.... thanks in advance

**Comment by aransfor** [comment]
also @jayich I think all of your comments were true, however to avoid similar conflicts in the future I would really like to have the same EXACT convention as the martinis group

File: lib/servers/shutterandswitch/Arduinoshutterandswitchserver.py [hunk 8]
```diff
@@ -16,81 +16,119 @@
...
+    @inlineCallbacks
+    def initServer(self):
+        print 'loading config info...',
+        self.reg = self.client.registry()
+        yield self.loadConfigInfo()
+        yield DeviceServer.initServer(self)
+
+    @inlineCallbacks
+    def loadConfigInfo(self):
+        """Load configuration information from the registry."""
```
Comment (jayich) [inline lib/servers/shutterandswitch/Arduinoshutterandswitchserver.py]: an example registry key/value pair here would be helpful.

**Comment by jayich** [comment]
LGTM.

**Comment by aransfor** [comment]
@theoriginaljuice waiting on you

**Comment by theoriginaljuice** [comment]
@aransfor LGTM.

**Pull request merged**